bluetooth/src/commonMain/kotlin/device/Device.kt

```diff
@@ -0,0 +1,323 @@
+/*
+ * Copyright 2022 Splendo Consulting B.V. The Netherlands
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+
+package com.splendo.kaluga.bluetooth.device
+
+import com.splendo.kaluga.base.utils.getCompletedOrNull
+import com.splendo.kaluga.logging.debug
+import com.splendo.kaluga.state.HotStateFlowRepo
+import com.splendo.kaluga.state.StateRepo
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.filterNot
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.transformLatest
+import kotlinx.coroutines.flow.updateAndGet
+import kotlinx.coroutines.launch
+import kotlin.coroutines.CoroutineContext
+
+typealias ConnectibleDeviceStateFlowRepo = StateRepo<ConnectibleDeviceState, MutableStateFlow<ConnectibleDeviceState>>
+
+interface Device {
+    val identifier: Identifier
+    val info: Flow<DeviceInfo>
+    val state: Flow<DeviceState>
+
+    suspend fun connect(): Boolean
+    fun handleConnected()
+    suspend fun disconnect()
+    suspend fun handleDisconnected()
+    fun rssiDidUpdate(rssi: Int)
+    fun advertisementDataAndRssiDidUpdate(advertisementData: BaseAdvertisementData, rssi: Int)
```

Comment:
This looks strange, why not keep only `advertisementDataDidUpdate` and when needed call `rssiDidUpdate` as we already have this method